lib/Onnxifi/GlowOnnxManager.h

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_ONNXIFI_GLOWONNXMANAGER_H
+#define GLOW_ONNXIFI_GLOWONNXMANAGER_H
+
+#include <unordered_set>
+
+namespace glow {
+namespace onnxifi {
+class BackendId;
+class Backend;
+class Event;
+class Graph;
+
+/// Singleton class for creating and destroying objects for the ONNX interface.
+/// GlowOnnxManager tracks objects it has created and can be used to check if
+/// if an ONNX interface object was created by glow or not.
+class GlowOnnxManager {
+public:
+  /// Get a reference to the GlowOnnxManager singleton. There should only ever
+  /// be one GlowOnnxManager.
+  static GlowOnnxManager &get();
+
+  // Disallow copying GlowOnnxManager to help enforce singleton pattern.
+  GlowOnnxManager(const GlowOnnxManager &) = delete;
+  GlowOnnxManager &operator=(const GlowOnnxManager &) = delete;
+
+  /// Add a new glow \p backendId to the set of valid backendIds.
+  void addBackendId(BackendId *backendId);
```

Comment:
we have typedefs for all those pointers, e.g., BackendIdPtr

Reply:
Yeah I was trying to just forward declare `BackendId` etc and not import Base.h unnecessarily here but I would have to re typedef those (not sure if that' even legit) or import Base.h just for those typedefs. I'll just import it.

Reply:
Orthogonal to the PR, but what's the rationale for those typedefs?  We don't generally do that for pointer types and I generally feel `XyzPtr` is less understandable than `Xyz*`.